### PR TITLE
Big viewer cleanup

### PR DESCRIPTION
**Pull request by myieye**
This is essentially ready for review, except for the fact that it's based off of an unmerged branch.
I was finding it somewhat depressing working on viewer code, so I just went for it. There's much more to do.

- Adds eslint to the viewer and fixes all the errors
- Fixes svelte-check
- Sorts options in option lists (I guess that could have been a seperate PR, but meh)
- Refactors how options-fields work. @hahn-kev might have some objections to that one (a0086dce9cf0f04d4373049bb7eb29da91ee66c5). It's kind of squashed in the middle...oops. The type handling deep down was very confusing for me (as well as somewhat unsafe and peculiar) and every consumer needed to handle all the mapping between options and values. I feel like I've made the components easy to use, though I did add a bit of complexity to the `MultiOptionEditor` and `SingleOptionEditor` that you might not be happy with.
- Fixed a long standing problem for VS Code users

Some other code debt I'm eager to get to eventually:
- Our folder structure doesn't make a ton of sense
- I think there's a bunch of dead field-config code
- The ProjectView component is massive

### CODE REVIEW

**Comment by hahn-kev** [comment]
Funny I was just thinking about how big the project view was. I would love to pull stuff out of that to make it smaller

**Review by hahn-kev — COMMENTED** [review]
over all I like the changes to the Option editors. Part of why they were odd is because of how they were used before when we used `FieldEditor` for everything, the way the binding worked there was weird and I kinda hacked it to get it working.

I'd like to see some defaults for some of the callbacks, but I'd be happy to accept it as it is now.

File: frontend/viewer/src/lib/entry-editor/inputs/CrdtMultiOptionField.svelte [hunk 1]
```diff
@@ -15,28 +13,19 @@
...
   let append: HTMLElement;
 
-
-  function asMultiSelectValues(values: any[]): string[] {
-    return values?.map(v => v.id) ?? [];
-  }
-  function asObjectValues(values: string[]) {
-    return values.map(v => ({id: v}));
-  }
+  $: sortedOptions = options.sort((a, b) => a.label.localeCompare(b.label));
```
[the remark this answers is not in the document]
Comment (myieye) [reply frontend/viewer/src/lib/entry-editor/inputs/CrdtMultiOptionField.svelte:20]: I also added this to `CrdtSingleOptionField`, which I had forgotten to apply sorting to

File: frontend/viewer/src/lib/entry-editor/object-editors/SenseEditor.svelte [hunk 2]
```diff
@@ -35,18 +31,25 @@
...
-                     wsType="first-analysis"
-                     options={$semanticDomains.map(sd => ({label: sd.code + ' ' + pickBestAlternative(sd.name, 'analysis', $writingSystems), value: sd.id}))}/>
+                    on:change
+                    bind:value={sense.semanticDomains}
+                    options={$semanticDomains}
+                    {readonly}
+                    id="semanticDomains"
+                    wsType="first-analysis"
+                    getValueId={sd => sd.id}
+                    getValueById={id => $semanticDomains.find(sd2 => sd2.id === id)}
```
[the remark this answers is not in the document]
Comment (myieye) [reply frontend/viewer/src/lib/entry-editor/object-editors/SenseEditor.svelte]: I hadn't thought of that. As discussed I'll re-attempt this in a follow up PR.

**Review by hahn-kev — APPROVED** [review]

**Pull request merged**